ad7768-1fmcz/ad77681.c

```diff
@@ -243,10 +313,321 @@ int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
+				{
+					checksum_byte = data_buffer[1] & 0xFF;
+					checksum_length = 3;
+				}	
+		}
+		
+		else																					// without status bit
+			{
+				if (dev->conv_len == AD77681_CONV_24BIT)											// 24bit ADC data + 8bit of CRC = 2 16bit frames
+					{
+						checksum_byte = data_buffer[1] & 0xFF;
+						checksum_length = 3;
+					}
+				
+				else																				// 16bit ADC data + 8bit of CRC = 2 16bit frames
+					{
+						checksum_byte = data_buffer[1] >> 8;
+						checksum_length = 2;
+					}	
+			}
+		
+		for (i = 0; i < checksum_length; i++)
+		{
+			if (i % 2)
+				checksum_buf[i] = data_buffer[i / 2] & 0xFF;
+			else
+				checksum_buf[i] = data_buffer[i / 2] >> 8;
+		}
+		
+		if (dev->crc_sel == AD77681_CRC)
+			checksum = ad77681_compute_crc8(checksum_buf, checksum_length, INITIAL_CRC_CRC8);
+		
+		else if (dev->crc_sel == AD77681_XOR)
+			checksum = ad77681_compute_xor(checksum_buf, checksum_length, INITIAL_CRC_XOR);
+		
+		
+		if (checksum != checksum_byte)	
+			ret = -1;
+		
+#ifdef CRC_DEBUG
+		
+		char ok[3] = { 'O', 'K' }, fault[6] = { 'F', 'A', 'U', 'L', 'T' };
+		sprintf(print_buf, "\n%x\t%x\t%x\tCRC %s", checksum_byte, checksum, status_byte, ((ret == -1) ? (fault) : (ok)));
+		print_debug(print_buf);	
+		
+#endif // CRC_DEBUG
+		
+	}	
+	
+	return ret;
+}
+
+/**
+ * Conversion from measured data to voltage
+ * @param dev - The device structure.
+ * @param measured_data - structure carying ADC measurement info
+ * @param voltage		- Array of voltage values
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_data_to_voltage(struct ad77681_dev *dev,
+	struct adc_data *measured_data,
+	double *voltage)
+{
+	int32_t converted_data = 0;
+	
+	for (measured_data->count = 0; measured_data->count < measured_data->samples; measured_data->count++)
+	{
+
+		if (measured_data->raw_data[measured_data->count] & 0x800000)
+			converted_data = (int32_t)((0xFF << 24) | measured_data->raw_data[measured_data->count]);
+		else
+			converted_data = (int32_t)((0x00 << 24) | measured_data->raw_data[measured_data->count]);
+				
+		voltage[measured_data->count] = (double)(((2.0 * (((double)(dev->vref)) / 1000.0)) / AD7768_FULL_SCALE)*converted_data); 	//       ((2*Vref)*code)/2^24	
+	}
+	return 0;	
+}
 
+/**
+ * Update ADCs sample rate depending on MCLK, MCLK_DIV and filter settings
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_update_sample_rate(struct ad77681_dev *dev)
+{
+	uint32_t sample_rate;
+	uint8_t mclk_div;
+	uint16_t osr;
+	
+	// Finding out MCLK divide
+	switch (dev->mclk_div)
+	{
+	case AD77681_MCLK_DIV_16:
+		mclk_div = 16;
+		break;
+	case AD77681_MCLK_DIV_8:
+		mclk_div = 8;
+		break;
+	case AD77681_MCLK_DIV_4:
+		mclk_div = 4;
+		break;
+	case AD77681_MCLK_DIV_2:
+		mclk_div = 2;
+		break;
+	default:
+		return -1;
+		break;
+	}
+	
+	
+	// Finding out decimation ratio
+	switch (dev->filter)
```

Comment:
similar to the `switch (dev->decimate)` statement, you should not have empty lines between cases.